logger.go

```diff
@@ -226,3 +246,15 @@ func (log *logger) Fatal(msg string, fields ...zapcore.Field) {
 func (log *logger) Facility() zapcore.Facility {
 	return log.fac
 }
+
+func (log *logger) clone() *logger {
```

Comment:
It looks like this is a straight up copy. We have a couple of other options here that avoid listing every field (and possibly missing out on a new field that's added later):
```go
func (log logger) clone() *logger {
  // Since this is a method on the value, a copy was already created and passed in as log.
  return &log
}
```

```go
func (log *logger) clone() *logger {
  copy := *log
  return &copy
}
```